File: admin-backend/admin-system/src/main/java/io/github/faustofan/admin/system/domain/model/SysOrg.java
package io.github.faustofan.admin.system.domain.model;

import io.github.faustofan.admin.shared.persistence.model.TenantAware;
import org.babyfish.jimmer.sql.*;
import java.util.List;

/**
 * 组织机构实体
 */
@Entity
@Table(name = "sys_org")
public interface SysOrg extends TenantAware {

    String name();

    @ManyToOne
    @OnDissociate(DissociateAction.DELETE)
    SysOrg parent();

    @OneToMany(mappedBy = "parent")
    List<SysOrg> children();

    @OneToMany(mappedBy = "org")
    List<SysUser> users();
}

File: src/main/kotlin/io/github/faustofan/admin/system/domain/entity/SysOrg.kt
package io.github.faustofan.admin.system.domain.entity

import io.github.faustofan.admin.common.model.TenantAware
import org.babyfish.jimmer.sql.*

@Entity
interface SysOrg : TenantAware {

    val name: String

    // 树形结构：父节点
    @ManyToOne
    @OnDissociate(DissociateAction.DELETE) // 父节点删除时，递归删除子节点
    val parent: SysOrg?

    // 树形结构：子节点
    @OneToMany(mappedBy = "parent")
    val children: List<SysOrg>

    // 关联：该部门下的用户
    @OneToMany(mappedBy = "org")
    val users: List<SysUser>
}